xmodule/library_content_block.py

```diff
@@ -723,57 +399,36 @@ def post_editor_saved(self, user, old_metadata, old_content):  # pylint: disable
-        See get_child_blocks() above
+    def format_block_keys_for_analytics(self, block_keys: list[tuple[str, str]]) -> list[dict]:
         """
-        return True
-
-    def get_content_titles(self):
+        Implement format_block_keys_for_analytics using the modulestore-specific legacy library original-usage system.
         """
-        Returns list of friendly titles for our selected children only; without
-        thi, all possible children's titles would be seen in the sequence bar in
-        the LMS.
+        def summarize_block(usage_key):
+            """ Basic information about the given block """
+            orig_key, orig_version = self.runtime.modulestore.get_block_original_usage(usage_key)
+            return {
+                "usage_key": str(usage_key),
+                "original_usage_key": str(orig_key.replace(version=None, branch=None)) if orig_key else None,
+                "original_usage_version": str(orig_version) if orig_version else None,
+            }
 
-        This overwrites the get_content_titles method included in x_module by default.
-        """
-        titles = []
-        for child in self.get_child_blocks():
-            titles.extend(child.get_content_titles())
-        return titles
-
-    @classmethod
-    def definition_from_xml(cls, xml_object, system):
-        children = []
-
-        for child in xml_object.getchildren():
+        result_json = []
+        for block_key in block_keys:
+            key = self.context_key.make_usage_key(*block_key)
+            info = summarize_block(key)
+            info['descendants'] = []
             try:
-                children.append(system.process_xml(etree.tostring(child)).scope_ids.usage_id)
-            except (XMLSyntaxError, AttributeError):
-                msg = (
-                    "Unable to load child when parsing Library Content Block. "
-                    "This can happen when a comment is manually added to the course export."
-                )
-                logger.error(msg)
-                if system.error_tracker is not None:
-                    system.error_tracker(msg)
-
-        definition = dict(xml_object.attrib.items())
-        return definition, children
-
-    def definition_to_xml(self, resource_fs):
-        """ Exports Library Content Block to XML """
-        xml_object = etree.Element('library_content')
-        for child in self.get_children():
-            self.runtime.add_block_as_child_node(child, xml_object)
-        # Set node attributes based on our fields.
-        for field_name, field in self.fields.items():  # pylint: disable=no-member
-            if field_name in ('children', 'parent', 'content'):
-                continue
-            if field.is_set_on(self):
-                xml_object.set(field_name, str(field.read_from(self)))
-        return xml_object
+                block = self.runtime.modulestore.get_item(key, depth=None)  # Load the item and all descendants
```

Comment:
I was going to ask the same thing, because the canonical way to iterate over children is just to use `self.get_chidlren()`. But I saw the "all descendants" - I think the intention here was something more optimized that loads many levels of decsendants at once? However, I don't really know why we care about that when 99% of problem banks will be one level deep.

Reply:
> an artifact of the fact that this used to be part of a class that had a reference to modulestore?

It's an artifact of that, plus me not remembering that `self.runtime.get_block` exists 😄 

Using `self.runtime.modulestore.get_item` is already an imperfect substitute, as it spit out blocks with version information in their `.children` keys, hence the new `.replace(version=None, branch=None)` clause.

I'll try `get_block`, and see if it lets me remove the version-stripping.

Reply:
This change didn't immediately work, so I'm going to punt on it for now